cli/tests/integration/cache_tests.rs

```diff
@@ -107,3 +107,49 @@ itest!(package_json_basic {
   copy_temp_dir: Some("package_json/basic"),
   exit_code: 0,
 });
+
+// Cache API put() regression tests.
+// See https://github.com/denoland/deno/issues/17299
+// Note: It is important that part1 runs before part2.
```

Comment:
If it's important for part1 to run before part2 then maybe these should be a single test? We could rewrite it using the `TestContextBuilder`.

Also, I'm not sure how this tests it because doesn't the `itest` macro use a fresh deno_dir each time? Also, doesn't Rust execute tests in parallel?

Reply:
> Also, I'm not sure how this tests it because doesn't the itest macro use a fresh deno_dir each time? 

It does

> Also, doesn't Rust execute tests in parallel?

Yup

+1 for rewriting it using `TestContextBuilder`

Reply:
TestContextBuilder is exactly what I wanted. Thanks!